ha/cli/commands.py

```diff
@@ -15,130 +15,81 @@
-                    help = "Manage cluster services and it's operations.")
-        action_subparser = service_parser.add_subparsers(
-            title="Service Actions",
-            dest="service_action")
-
-        start_parser = action_subparser.add_parser("start",
-            help = "Start Service")
-        start_parser.add_argument("service_name",
-            help="Service name", action="store")
-        start_parser.add_argument("--node",
-            help="Provide node name to start service on particular node")
-
-        stop_parser = action_subparser.add_parser("stop",
-            help = "Stop Service")
-        stop_parser.add_argument("service_name",
-            help="Service name", action="store")
-        stop_parser.add_argument("--node",
-            help="Provide node name to stop service on particular node")
-
-        status_parser = action_subparser.add_parser("status",
-            help = "Status of service")
-        status_parser.add_argument("service_name",
-            help="Service name", action="store")
-        status_parser.add_argument("--node",
-            help="Provide node name to get status of service.")
-
-class SupportBundleCommand(Command):
-    @staticmethod
-    def add_args(component_parser):
-        """
-        Add parser for support bundle command
-
-        Args:
-            component_parser (<class 'argparse._SubParsersAction'>): argparse to add subparser.
-        """
-        bundle_parser =  component_parser.add_parser("support_bundle",
-                                help = "Manage support bundle operation like create.")
-        action_subparser = bundle_parser.add_subparsers(
-            title="Support Bundle",
-            dest="bundle_action")
-
-        create_parser = action_subparser.add_parser("create",
-            help = "Create Support Bundle")
-        create_parser.add_argument("id",
-            help="Bundle id", action="store")
-        create_parser.add_argument("path",
-            help="Bundle Path", action="store")
\ No newline at end of file
+
+    def __init__(self):
+        self.module_name = None
+        self.operation_name = None
+        self.options = None
+        self.cmd_factory = cmdFactory()
+
+        # Modules where executor classes are defined
+        self.execute = [commandExecutor, clusterExecutor, nodeExecutor, serviceExecutor, storagesetExecutor]
+
+
+    def parse(self, args):
+        """ Parse the CLI string to identify parameters"""
+
+        try:
+            self.module_name = args[0]
+            self.operation_name = args[1]
+            self.options = args[2:]
+
+        except Exception:
+
+            print(CLIUsage.usage())
+            if self.module_name != "-h" and self.module_name != "--help":
+                raise HAInvalidCommand("Invalid parameters passed; refer to help for details")
+            else:
+                return False
+        return True
+
+
+    def get_class(self, commandExecutor):
```

Comment:
This is not required and self.execute is also not required. Use dictionary created in CommandFactory.

Reply:
Done